TFNetworkLayer.py

```diff
@@ -2953,8 +2957,15 @@ def ceildiv(a, b):
 
   @classmethod
   def _get_out_type_from_opts(cls, n_out, filter_size, padding, strides=1, dilation_rate=1, sources=(),
-                              input_expand_dims=0, input_add_feature_dim=False, input_split_feature_dim=None, **kwargs):
-    shape = [None] * len(filter_size) + [n_out]
+                              input_expand_dims=0, input_add_feature_dim=False, input_split_feature_dim=None,
+                              auto_use_channel_first=False,
+                              **kwargs):
+    data = get_concat_sources_data_template(sources)
+    # The output format is the same as the input.
+    if data.feature_dim_axis == 1:
```

Comment:
This is wrong. I think/guess, you actually want to check if the input data is also channel first. Then also check that, and not if `feature_dim_axis == 1` (not the same thing).

Reply:
Also, the comment should be extended. Say that you want channel first if the input is also channel first.
But actually, I think the code is simpler if you just remove the code here completely, and leave it like it was originally (`shape = [None] * len(filter_size) + [n_out]`), and then just handle it in the already existing code in the end.